### PR TITLE
Add libsharp

### PR DESCRIPTION
**Pull request by moxcodes**
## Proposed changes

Add libsharp Setup and Find files to cmake system. Small clarification change
to dependency documentation.

### Types of changes:

- [ ] New feature

### Component:

- [ ] Build system
- [ ] Documentation

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: cmake/FindLibsharp.cmake [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+# Find libsharp: https://github.com/Libsharp/libsharp
+# If not in one of the default paths specify -D LIBSHARP_ROOT=/path/to/libsharp
+# to search there as well.
+
+include (CheckCXXSourceRuns)
+
+# find the ssht include directory
```
Comment (wthrowe) [inline cmake/FindLibsharp.cmake]: libsharp

**Comment by moxcodes** [comment]
fixed.
Cheers!

**Comment by moxcodes** [comment]
Updated with a few more links that are needed to make the library actually work in practice.

**Comment by hrueter** [comment]
The dependency on libsharp should be added on the installation page in the SpECTRE documentation.

**Comment by moxcodes** [comment]
@hrueter good point! I've added the appropriate documentation lines.

**Comment by kidder** [comment]
this needs a rebase

**Review by nilsdeppe — APPROVED** [review]
Could you verify that this works on BlueWaters and Stampede2?

**Comment by moxcodes** [comment]
libsharp builds on Stampede2, I'll work on checking BlueWaters too.

**Comment by moxcodes** [comment]
With the compile flags we require (-openmp, -mpi), libsharp also builds on Blue Waters via spack.

**Comment by nilsdeppe** [comment]
Awesome, I'm happy with this to be merged then :)

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
@wthrowe please look at

**Review by wthrowe — APPROVED** [review]

**Pull request merged**